components/rename_channel_modal/rename_channel_modal.tsx

```diff
@@ -34,61 +36,45 @@ const holders = defineMessages({
     },
 });
 
-export class RenameChannelModal extends React.PureComponent {
-    static propTypes = {
-
-        /**
-         * react-intl helper object
-         */
-        intl: intlShape.isRequired,
-
-        /**
-         * Function that is called when modal is hidden
-         */
-        onHide: PropTypes.func.isRequired,
-
-        /**
-         * Object with info about current channel
-         */
-        channel: PropTypes.object.isRequired,
-
-        /**
-         * Object with info about current team
-         */
-        team: PropTypes.object.isRequired,
-
-        /**
-         * String with the current team URL
-         */
-        currentTeamUrl: PropTypes.string.isRequired,
-
-        /*
-         * Object with redux action creators
-         */
-        actions: PropTypes.shape({
-
-            /*
-             * Action creator to patch current channel
-             */
-            patchChannel: PropTypes.func.isRequired,
-        }).isRequired,
+type Props = {
+    intl: IntlShape;
+    onHide: () => void;
+    channel: Channel;
+    team: Team;
+    currentTeamUrl: string;
+    actions: {
+        patchChannel: (channelId: string, patch: Channel) => Promise<{ data: Channel, error: Error }>;
     };
+}
+
+type State = {
+    displayName: string;
+    channelName: string;
+    serverError?: string;
+    urlErrors: JSX.Element[];
+    displayNameError: JSX.Element | string;
```

Comment:
There's a React provided type called `React.ReactNode` that includes both these types and any others that you can render to the DOM.